### PR TITLE
fix(notifications): display proper type in order notifications

### LINKED ISSUE #3741: [1.60] Cancelled/Expired/Presigned notifications for Limit/TWAP orders
Continuation of the #3731 task: 
1. Cancelled Limit/TWAP orders are labeled as SWAP
limit 3
limit
limit1
2. Expired Limit/TWAP orders are labeled as SWAP
image

3. Failed ETH order has also a bit different notification (maybe leave as it is, not sure)
image

4. Presigned Limit/TWAP orders: 
image

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #3741

`computeOrderSummary()` used to have hardcoded `Swap`, I've replaced it by `getUiOrderType()` value.

  # To Test

See #3741

P.S. The bug was not introduced in 1.60 version, it was there before

### CODE REVIEW

File: apps/cowswap-frontend/src/common/updaters/orders/utils.ts [hunk 1]
```diff
@@ -19,55 +20,53 @@ export function computeOrderSummary({
...
-      // We only have the API order info, let's at least use that
-      summary = `Swap ${sellToken} for ${buyToken}`
-    }
-
-    if (owner && receiver && receiver !== owner) {
-      summary += ` to ${shortenAddress(receiver)}`
-    }
+  if (!orderFromStore && !orderFromApi) return undefined
+
+  const buyToken = orderFromApi?.buyToken || orderFromStore?.buyToken
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/common/updaters/orders/utils.ts:25]: In the code above we explicitly exit from function when both orders are falsy, it means that here at least one of them is truthy.
But typescript is not so smart, so I had to cast to string some consts.

**Review by alfetopito — APPROVED** [review]
Expiration worked fine

image

**Review by anxolin — APPROVED** [review]
Looks good to me 
image

**Comment by elena-zh** [comment]
Hey @shoom3301 , cancelled pop-up for the Limit orders says Swap (instead of the Limit order)
limit cancelled

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed it as well

File: apps/cowswap-frontend/src/legacy/state/orders/middleware/batchCancelOrdersPopup.ts [hunk 2]
```diff
@@ -2,25 +2,33 @@ import { orderAnalytics } from '@cowprotocol/analytics'
...
+import { deserializeOrder } from '../utils/deserializeOrder'
 
-export function batchCancelOrdersPopup(store: MiddlewareAPI<Dispatch, AppState>, orders: BaseOrder[]) {
+export function batchCancelOrdersPopup(store: MiddlewareAPI<Dispatch, AppState>, orders: OrderObject[]) {
   // construct Cancelled Order Popups for each Order
   orders.forEach((order) => {
     const popup = _buildCancellationPopup(order)
-    const orderType = order.composableCowInfo ? 'TWAP' : order.class
+
+    const orderType = order.order.composableCowInfo ? 'TWAP' : order.order.class
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/state/orders/middleware/batchCancelOrdersPopup.ts:19]: Use `getUiOrderType`? https://github.com/cowprotocol/cowswap/blob/a2f8def2f203301885e4548b49f73336abbabe21/apps/cowswap-frontend/src/utils/orderUtils/getUiOrderType.ts#L33-L32

Comment (shoom3301) [reply apps/cowswap-frontend/src/legacy/state/orders/middleware/batchCancelOrdersPopup.ts:19]: https://github.com/cowprotocol/cowswap/pull/3901

🙈

**Pull request merged**